pkg/apis/wsecurity/v1alpha1/guardianSpec.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+type Ctrl struct {
+	Alert bool `json:"alert"` // If true, use SessionDataConfig to identify alerts
+	Block bool `json:"block"` // If true, block on alert.
+	Learn bool `json:"learn"` // If true, and no alert identified, report piles
+	Force bool `json:"force"` // If true, learning is done even when alert identified, report piles
+	Auto  bool `json:"auto"`  // If true, use learned SessionDataConfig rather than configured SessionDataConfig
+}
+
+type GuardianSpec struct {
+	Configured *SessionDataConfig `json:"configured"`        // configrued criteria
+	Learned    *SessionDataConfig `json:"learned,omitempty"` // Learned citeria
+	Control    *Ctrl              `json:"control"`           // Control
+}
+
+// AutoActivate is a Guardian operation mode that is useful for security automation of new services
+func (g *GuardianSpec) AutoActivate() {
```

Comment:
Would `ActivateAll()` be a better name? (`Auto` feels a bit vague in terms of what the method is doing)

Reply:
ActivateAll is not a better name here :) since we are not activating blocking for example. 
It is a setup suitable for maximal automation and requires no user intervention. 
A good start for new services, while in development, and possibly when they start production and before you know what micro-rules to set. 

You can suggest a name :)

Reply:
Changed to SetToMaximalAutomation()

Reply:
works for me